starlette/datastructures.py

```diff
@@ -123,76 +123,76 @@ class Headers(typing.Mapping[str, str]):
     An immutable, case-insensitive multidict.
     """
 
-    def __init__(self, raw_headers=None) -> None:
+    def __init__(self, raw_headers: typing.Any = None) -> None:
         if raw_headers is None:
-            self._list = []
+            self._list = []  # type: typing.List
         else:
             for header_key, header_value in raw_headers:
                 assert isinstance(header_key, bytes)
                 assert isinstance(header_value, bytes)
                 assert header_key == header_key.lower()
             self._list = raw_headers
 
-    def keys(self):
+    def keys(self) -> typing.List[str]:  # type: ignore
         return [key.decode("latin-1") for key, value in self._list]
 
-    def values(self):
+    def values(self) -> typing.List[str]:  # type: ignore
         return [value.decode("latin-1") for key, value in self._list]
 
-    def items(self):
+    def items(self) -> StrPairs:  # type: ignore
         return [
             (key.decode("latin-1"), value.decode("latin-1"))
             for key, value in self._list
         ]
 
-    def get(self, key: str, default: str = None):
+    def get(self, key: typing.Any, default: typing.Any = None) -> str:
         try:
             return self[key]
         except KeyError:
-            return default
+            return default  # type: ignore
 
-    def getlist(self, key: str) -> typing.List[str]:
+    def getlist(self, key: typing.Any) -> typing.List[str]:
         get_header_key = key.lower().encode("latin-1")
         return [
             item_value.decode("latin-1")
             for item_key, item_value in self._list
             if item_key == get_header_key
         ]
 
-    def mutablecopy(self):
+    def mutablecopy(self) -> "MutableHeaders":
         return MutableHeaders(self._list[:])
 
-    def __getitem__(self, key: str):
+    def __getitem__(self, key: typing.Any) -> str:
         get_header_key = key.lower().encode("latin-1")
         for header_key, header_value in self._list:
             if header_key == get_header_key:
                 return header_value.decode("latin-1")
         raise KeyError(key)
 
-    def __contains__(self, key: str):
+    def __contains__(self, key: typing.Any) -> bool:
```

Comment:
yes, the key should be a string. The thing is when I set `key: string` I have the following errors:

```
error: Argument 1 of "__contains__" incompatible with supertype "Mapping"
error: Argument 1 of "__contains__" incompatible with supertype "Container"
```

What I understand is that I have to override the signatures for the parent classes, maybe doing something like:

```
@typing.overload
def __contains__(self, key: object ) -> bool: ...
@typing.overload
def __contains__(self, key: typing.Any) -> bool: ...
def __contains__(self, key: str) -> bool:
      # implementation
```

This kind of situations happens on several signatures when we want to override inheritance methods. This is why I am using `typing.Any` for this cases.

Reply:
Okay, Ive bumped into that before - that's tedious.

I expect that there may be more correct ways to deal with it by eg. subclassing a generic type, but I'm not sure. Personally I think we should consider just switching off mypy's "incompatible with supertype" errors. We're doing a perfectly correct thing, since the type we're using is a subset of the parent.

Slight aside: What arguments are you using for running mypy, and are there other errors still currently remaining? (At some point it'd be good to add it to the linting perhaps)

Reply:
Yes, I was expecting the same. I am using `mypy starlette --disallow-untyped-defs`, the same that is described on the #7 

Mostly all errors are related with `incompatible with supertype` on `datastructure.py`. 
I've finished with all the corrections in `requests.py`. 

Yes, after all the annotations will be fixed, should be good idea to add the `"mypy"` stage to the pipeline.

Reply:
I asked about it and whether there is a flag to switching off mypy's " incompatible with supertype" errors on `mypy`. The answer was: 

`There is no global flag for this, because Liskov principle is like one of cornerstones of the type system. You can silence the error by # type: ignore of course.`

You can see it [here](https://github.com/python/mypy/issues/5633)

Reply:
Okay, sticking with `typing.Any` there make sense 👍

Reply:
Sounds good!